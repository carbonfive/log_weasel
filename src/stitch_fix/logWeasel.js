import { ulid } from 'ulid';

let globalKey;

const LogWeasel = {
  init: (key) => {
    if (typeof key === 'undefined') {
      throw new TypeError('LogWeasel.init requires a key argument');
    }

    globalKey = key;
  },

  generateId: () => {
    const formattedKey = `${globalKey}-JS`;

    return `${ulid()}-${formattedKey}`;
  },
};

export { LogWeasel };
export default LogWeasel;
