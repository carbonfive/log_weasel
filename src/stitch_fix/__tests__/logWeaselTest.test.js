import { LogWeasel } from '../logWeasel';

describe('init', () => {
  it('needs a key', () => {
    expect(() => LogWeasel.init()).toThrow();
  });
});

describe('generateId', () => {
  beforeAll(() => {
    LogWeasel.init('WEASEL');
  });

  it('returns a Log Weasel ID', () => {
    expect(LogWeasel.generateId()).toMatch(/^[0-9A-Z]{26}-WEASEL-JS$/);
  });

  it('returns a different Log Weasel ID each time it is called', () => {
    const firstId = LogWeasel.generateId();
    const secondId = LogWeasel.generateId();

    expect(firstId).not.toEqual(secondId);
  });
});

