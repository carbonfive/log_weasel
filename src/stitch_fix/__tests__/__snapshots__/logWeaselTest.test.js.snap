// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init throws an error if key is not passed 1`] = `"LogWeasel.init requires a key argument"`;
